daemon/src/model/cfd.rs

```diff
@@ -683,6 +783,72 @@ impl Cfd {
         Ok(signed_commit_tx)
     }
 
+    pub fn cet(&self) -> Result<Result<Transaction, NotReadyYet>> {
+        let (dlc, attestation) = match self.state.clone() {
+            CfdState::OpenCommitted {
+                dlc,
+                cet_status: CetStatus::Ready(attestation),
+                ..
+            }
+            | CfdState::PendingCet {
+                dlc,
+                cet_status: CetStatus::Ready(attestation),
+                ..
+            } => (dlc, attestation),
+            CfdState::OpenCommitted { .. }
+            | CfdState::Open { .. }
+            | CfdState::PendingCommit { .. } => {
+                return Ok(Err(NotReadyYet));
+            }
+            _ => bail!("Cannot publish CET in state {}", self.state.clone()),
+        };
+
+        let cets = dlc
+            .cets
+            .get(&attestation.id.0)
+            .context("Unable to find oracle event id within the cets of the dlc")?;
```

Comment:
Can we push such low-level code down into the respective types like `Dlc` please?